### PR TITLE
Install Guide

### PR DESCRIPTION
**Pull request by karl-cardenas-coding**
This PR adds an install guide to help newer users. The creation of `docs` folder is also added to help the maintainer team add more documentation down the road.

Use this for the [preview link](https://github.com/spectrocloud-labs/validator/blob/docs-update/docs/install.md)

### CODE REVIEW

File: docs/tutorial.md [hunk 1]
```diff
@@ -0,0 +1,290 @@
...
+
+
+## Prerequisites
+
+- An AWS account.
+
+- AWS credentials with [AdministratorAccess](https://docs.aws.amazon.com/aws-managed-policy/latest/reference/AdministratorAccess.html) policy attached. You can create a new user with this policy and use the credentials with the AWS plguin. Refer to the [AWS IAM User](https://docs.aws.amazon.com/IAM/latest/UserGuide/id_users_create.html) documentation for more details.
+
+> [!IMPORTANT]
+> You can use a different policy, but ensure that the policy has the required permissions to check the resources required by the plugin. Refer to the AWS plugin documentation for more details.
```
Comment (TylerGillson) [inline docs/tutorial.md:13]: "has the required permissions to check" -> "has the permissions required to check"

File: docs/tutorial.md [hunk 2]
```diff
@@ -0,0 +1,290 @@
...
+
+
+
+Create a **values.yaml** containing the Validator configuration. Use the command below to download the default **values.yaml** file.
+
+```shell
+wget https://raw.githubusercontent.com/spectrocloud-labs/validator/main/chart/validator/values.yaml
+```
+
+Use a text editor to edit the **values.yaml** file and change navigate down to the `plugins` section and locate the `AWS` plugin.
```
Comment (TylerGillson) [inline docs/tutorial.md:76]: "and change navigate down" -> ", navigate down"

File: docs/tutorial.md [hunk 3]
```diff
@@ -0,0 +1,290 @@
...
+values: |-
+# Remainder of the file omitted for brevity
+```
+
+The AWS plugin requires credentials to access your AWS account. Uncomment the `auth` section and add your AWS credentials. By default the auth section contains a an empty `{}` block. Replace the empty block with the following configuration. Replace the `accessKeyId` and `secretAccessKey` with your AWS credentials. 
+
+```yaml
+auth:
+# Leave secret undefined for implicit auth (node instance role, IMDSv2, etc.)
+secret:
```
Comment (TylerGillson) [inline docs/tutorial.md:93]: L94-101 need to be indented

File: docs/tutorial.md [hunk 4]
```diff
@@ -0,0 +1,290 @@
...
+```
+
+Now that you have Validator installed, you can create a validation. A validation is a custom resource that contains the configuration for the validation. Validator will create a `ValidationResult` custom resource that contains the result of the validation.
+
+## Create a Validation
+
+The next step is to create a validation configuration. Each plugin contains a set of example validation configurations. You can use the example configurations as a starting point for your validation. For this guide, you will use the [`awsvalidator-spectro-cloud-base`](https://github.com/spectrocloud-labs/validator-plugin-aws/blob/main/config/samples/awsvalidator-iam-role-spectro-cloud-base.yaml) configuration. 
+
+> [!NOTE]
+> Check ou the AWS [config samples](https://github.com/spectrocloud-labs/validator-plugin-aws/tree/main/config/samples) directory for more examples.
```
Comment (TylerGillson) [inline docs/tutorial.md:132]: "Check ou the" -> "Check out the"

File: docs/tutorial.md [hunk 5]
```diff
@@ -0,0 +1,290 @@
...
+
+
+Create a file named `validation.yaml` and copy the contents of the [`awsvalidator-spectro-cloud-base`](https://github.com/spectrocloud-labs/validator-plugin-aws/blob/main/config/samples/awsvalidator-iam-role-spectro-cloud-base.yaml) Use the following command to download the file and save it as `validation.yaml`.
+
+```shell
+wget https://raw.githubusercontent.com/spectrocloud-labs/validator-plugin-aws/main/config/samples/awsvalidator-iam-role-spectro-cloud-base.yaml --output-document validation.yaml
+```
+
+Review the contents of the `validation.yaml` file. The file contains the configuration for the validation. The `spec` section contains the configuration for the validation.  Change the `spec` section to match your requirements. For example, you can change the `defaultRegion` to match your prefered AWS region.
```
Comment (TylerGillson) [inline docs/tutorial.md:141]: Delete "The file contains the configuration for the validation." ?

File: docs/tutorial.md [hunk 6]
```diff
@@ -0,0 +1,290 @@
...
+    Message:               One or more required IAM permissions was not found, or a condition was not met
+    Status:                False
+    Validation Rule:       validation-SpectroCloudRole
+    Validation Type:       aws-iam-role-policy
+  Sink State:              N/A
+  State:                   Failed
+Events:                    <none>
+```
+
+If the validation fails, you can review the `Failures` section of the `Conditions` section to determine the cause of the failure. In this example, the IAM role is missing the required permissions. You can update the IAM role to add the missing permissions and rerun the validation.
```
Comment (TylerGillson) [inline docs/tutorial.md:215]: "of the `Conditions` section to" -> "of each `Condition` in its `Status` to"

Comment (TylerGillson) [reply docs/tutorial.md:215]: "and rerun the validation" -> "and wait 30s for the validation to be revalidated"

File: docs/tutorial.md [hunk 7]
```diff
@@ -0,0 +1,290 @@
...
+    Status:                True
+    Validation Rule:       validation-SpectroCloudRole
+    Validation Type:       aws-iam-role-policy
+  Sink State:              N/A
+  State:                   Succeeded
+Events:                    <none>
+```
+
+
+The validation results are hashed and result events are only updated if the result has changed. In this example, the validation check was successful and the `ValidationResult` CR was updated with the result of the validation.
```
Comment (TylerGillson) [inline docs/tutorial.md:256]: "result events are only updated" -> "result events are only emitted"

A brief intro to sinks would be helpful before this sentence. The ValidationResult CRs are always updated with whatever the latest details are. But the validation controller will only re-emit events to your configured sink if the hash changes.

File: docs/tutorial.md [hunk 8]
```diff
@@ -0,0 +1,290 @@
...
+    Validation Type:       aws-iam-role-policy
+  Sink State:              N/A
+  State:                   Failed
+Events:                    <none>
+```
+
+If the validation fails, you can review the `Failures` section of the `Conditions` section to determine the cause of the failure. In this example, the IAM role is missing the required permissions. You can update the IAM role to add the missing permissions and rerun the validation.
+
+
+The Validator will continouslly re-issue a validation cand update the `ValidationResult` CR with the result of the validation. You can use the following command to get the status of the validation. Once you addressed the cause of the failure, the validation will pass.
```
Comment (lennessyy) [inline docs/tutorial.md:218]:
```suggestion
The Validator will continouslly re-issue a validation and update the `ValidationResult` CR with the result of the validation. You can use the following command to get the status of the validation. Once you addressed the cause of the failure, the validation will pass.
```

File: docs/tutorial.md [hunk 9]
```diff
@@ -0,0 +1,290 @@
...
+```
+
+
+The validation results are hashed and result events are only updated if the result has changed. In this example, the validation check was successful and the `ValidationResult` CR was updated with the result of the validation.
+
+
+
+## Cleanup
+
+To delete the Validator installation, use the following command. This command will delete the Validator and all deployed resources including the `ValidationResult` CRs.
```
Comment (lennessyy) [inline docs/tutorial.md:262]:
```suggestion
To delete the Validator installation, use the following command. This command deletes the Validator and all deployed resources including the `ValidationResult` CRs.
```

**Review by lennessyy — APPROVED** [review]

File: docs/tutorial.md (hunk 8 above)
Comment (TylerGillson) [reply docs/tutorial.md:218]: Once you've addressed ?

**Comment by karl-cardenas-coding** [comment]
@TylerGillson ready for your re-review. All feedback has been incorporated 😄

**Review by TylerGillson — CHANGES_REQUESTED** [review]
Everything looks great, except for the AWS plugin auth configuration. That config changed last week due to a security issue that we identified.

**Review by TylerGillson — APPROVED** [review]

**Pull request merged**